vcbuild.ps1

```diff
@@ -0,0 +1,570 @@
+  if $? -eq 0 return
+
+  $path-item | findstr /r /c:"test\\addons\\[0-9].*_.*\.cc"
+  if $? -eq 0 return
+
+  set "localcppfilelist=%localcppfilelist% %1"
+  return
+}
+
+
+function jslint
+{
+  if defined jslint_ci & $jslint-ci
+  if not defined jslint return
+  if not exist tools\eslint\lib\eslint.js $ $no-lint
+  echo running jslint
+  %config%\node tools\eslint\bin\eslint.js --cache --rule "linebreak-style: 0" --rulesdir=tools\eslint-rules benchmark lib test tools
+  return
+
+  $jslint-ci = {
+    echo running jslint-ci
+    %config%\node tools\jslint.js -J -f tap -o test-eslint.tap benchmark lib test tools
+    return
+  }
+
+  $no-lint = {
+    echo Linting is not available through the source tarball.
+    echo Use the git repo instead: $ git clone https://github.com/nodejs/node.git
+    return
+  }
+}
+
+
+function create-msvs-files-failed
+{
+  echo Failed to create vc project files.
+  exit(0)
+}
+
+
+function help
+{
+  echo vcbuild.bat [debug/release] [msi] [test-all/test-uv/test-inspector/test-internet/test-pummel/test-simple/test-message] [clean] [noprojgen] [small-icu/full-icu/without-intl] [nobuild] [sign] [x86/x64] [vc2015] [download-all] [enable-vtune] [lint/lint-ci]
+  echo Examples:
+  echo   vcbuild.bat                : builds release build
+  echo   vcbuild.bat debug          : builds debug build
+  echo   vcbuild.bat release msi    : builds release build and MSI installer package
+  echo   vcbuild.bat test           : builds debug build and runs tests
+  echo   vcbuild.bat build-release  : builds the release distribution as used by nodejs.org
+  echo   vcbuild.bat enable-vtune   : builds nodejs with Intel VTune profiling support to profile JavaScript
+  exit(0)
+}
+
+
+# ***************
+#   Subroutines
+# ***************
+
+
+function getnodeversion
+{
+  $NODE_VERSION=
+  $TAG=
+  $FULLVERSION=
+
+  for /F "usebackq tokens=*" %%i in (`python "%~dp0tools\getnodeversion.py"`) do $NODE_VERSION=%%i
+  if not defined NODE_VERSION (
+    echo Cannot determine current version of Node.js
+    exit /b 1
+  )
+
+  if not defined DISTTYPE $DISTTYPE=release
+  if "%DISTTYPE%" -eq "release" (
+    $FULLVERSION=%NODE_VERSION%
+    return
+  )
+  if "%DISTTYPE%" -eq "custom" (
+    if not defined CUSTOMTAG (
+      echo "CUSTOMTAG is not set for DISTTYPE=custom"
+      exit /b 1
+    )
+    $TAG=%CUSTOMTAG%
+  )
+  if not "%DISTTYPE%" -eq "custom" (
+    if not defined DATESTRING (
+      echo "DATESTRING is not set for nightly"
+      exit /b 1
+    )
+    if not defined COMMIT (
+      echo "COMMIT is not set for nightly"
+      exit /b 1
+    )
+    if not "%DISTTYPE%" -eq "nightly" (
+      if not "%DISTTYPE%" -eq "next-nightly" (
+        echo "DISTTYPE is not release, custom, nightly or next-nightly"
+        exit /b 1
+      )
+    )
+    $TAG=%DISTTYPE%%DATESTRING%%COMMIT%
+  )
+  $FULLVERSION=%NODE_VERSION%-%TAG%
+}
+
+
+:exit
+if not defined DISTTYPEDIR $DISTTYPEDIR=%DISTTYPE%
+goto :EOF
```

Comment:
Can I suggest adding this file in a second commit (or PR)? The changes don’t seem related?